Userland/Libraries/LibGfx/ImageFormats/TIFFLoader.cpp

```diff
@@ -521,14 +521,24 @@ class TIFFLoadingContext {
         VERIFY_NOT_REACHED();
     }
 
+    ErrorOr<void> set_next_ifd(u32 ifd_offset)
+    {
+        if (ifd_offset != 0) {
+            if (ifd_offset < TRY(m_stream->tell()))
```

Comment:
Somewhat nitty: API-wise it feels a bit wobbly to me that this function gets the offset passed in, but compares it to the current stream position, implying that it gets read right before this function gets called every time. If that's the case, I'd either have both the read() and the tell() in this function, or both outside and pass both values in, to make sure they're from a consistent stream state.

Reply:
I'm not a huge fan either, I already tried my best on that. The issue is that they can come either from the `next_ifd` field of an existing `ifd` or from a tag (e.g. ExifIFD). They are, unfortunately, not read from the stream in the same way, making the factorization impossible.
And while it makes sense to call `tell()` in `read_next_idf_offset()`, calling it in the `subifd_handler` lambda doesn't make more sense, and it puts distance between the call and the usage of the value.